light/provider/http/http.go

```diff
@@ -68,7 +68,7 @@ func (p *http) LightBlock(ctx context.Context, height int64) (*types.LightBlock,
 		return nil, err
 	}
 
-	vs, err := p.validatorSet(ctx, h)
+	vs, err := p.validatorSet(ctx, &sh.Height)
```

Comment:
how's this better than using `h`?

Reply:
I'm thinking what happens if h is `nil` i.e. grab the latest height but the height for the latest validator set is different to the height of the latest commit (especially as a commit aka a signed header can only be formed at the following height)

Reply:
but h can't be nil

Reply:
really? what about this in `validateHeight`
```
if height == 0 {
	h = nil
}
return h, nil
```

Reply:
ah, right.